clients/src/main/java/org/apache/kafka/common/protocol/Protocol.java

```diff
@@ -1180,7 +1193,10 @@
     public static final Schema INIT_PRODUCER_ID_REQUEST_V0 = new Schema(
             new Field("transactional_id",
                     NULLABLE_STRING,
-                    "The transactional id whose producer id we want to retrieve or generate.")
+                    "The transactional id whose producer id we want to retrieve or generate."),
+            new Field("transaction_timeout_ms",
```

Comment:
This is a bacward incompatible change. The `INIT_PRODUCER_ID_REQUEST_V0` is already used without the `transaction_timeout_ms` field for the idempotent producer. I think we need not bump the version here because the RPC is only used if idempotence is enabled, and the existing code has been out for less than a week, so the risk is low of a new feature being used in production when it is turned off by default.